group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -703,6 +723,18 @@ public void run() {
                             // Add the response to the deferred queue.
                             if (!future.isDone()) {
                                 context.deferredEventQueue.add(offset, this);
+                                timer.add(new TimerTask(defaultWriteTimeout.toMillis()) {
+                                    @Override
+                                    public void run() {
+                                        if (!future.isDone()) {
+                                            scheduleInternalOperation(
+                                                "LogAppendEvent(name=" + name + ", tp=" + tp + ")",
+                                                tp,
+                                                () -> complete(new TimeoutException("Log append event " + name + "timed out for TopicPartition " + tp))
```

Comment:
nit: `"CoordinatorWriteEvent " + name + " timed out after " + timeoutMs + "ms"`.

Reply:
Another thing, we should use `org.apache.kafka.common.errors.TimeoutException`. Could we also ensure this in the test `testScheduleWriteOpWhenWriteTimesOut`?